### DISCUSSION TITLE
Sonoff 4ch Pro 2 switches won't work at the same time.

### DISCUSSION THREAD

**Discussion by xtremedays** [RavenSystem/esp-homekit-devices#1679] [Q&A]
Hi all,

Bit of a weird one... This 4ch Pro R3 sonoff I have with temp sensor on GPIO3, works perfectly individually all the switches work, does what its supposed to do. But if I try to turn on 2 switches at the same time, which ever was turned on first works, the other tile says 'ON' but doesn't activate the relay. If I switch the 1st switch off, the second relay then activates.... any thoughts? JSON as follows:

{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":15}]},"1":{"r":[{"g":15,"v":1}]},"b":[{"g":14}],"s":0},{"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0}],"s":0},{"0":{"r":[{"g":5}]},"1":{"r":[{"g":5,"v":1}]},"b":[{"g":9}],"s":0},{"0":{"r":[{"g":4}]},"1":{"r":[{"g":4,"v":1}]},"b":[{"g":10}],"s":0},{"t":22,"g":3,"n":3}]}

TIA

**Comment by xtremedays** [comment]
Anyone please? Still having issue with this. One automation didn't work last night because relay 3 was accidentally on, causing relay 1 not to activate at the set time.

**Comment by xtremedays** [comment]
I still haven’t come up with a solution for this. Can anyone help!?

**Comment by ceg117** [comment]
Although I want to help you, but I don't have the same equipment, can not detect the circuit situation and line direction, so dare not answer your question rashly, I suggest you first figure out the circuit diagram, and then configure JSON according to the GPIO definition, all software is generated by adapting to the hardware, I think this is why no one answers you.
